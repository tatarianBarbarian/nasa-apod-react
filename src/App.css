.controls {
    padding-top: 20px;
    text-align: center;
    margin-bottom: 20px;
}

.controls > *:not(:last-child) {
    margin-right: 8px;
    margin-bottom: 10px;
    }

.viewzone {
    display: grid;
    grid-template-columns: 45% minmax(400px, 600px);
    column-gap: 20px;
    justify-content: center;
}

.description {
    font-size: 18px;
    line-height: 1.5;
}
