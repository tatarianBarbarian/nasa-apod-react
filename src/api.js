import axios from 'axios';

const API_URL = '/.netlify/functions/nasa-apod-api';

const fetchApodData = async ({queryKey}) => {
    try {
        const {data} = await axios.get(API_URL, {
            params: {
                date: queryKey[1]
            }
        });
        
        if (data.code) {
            throw new Error(`Error ${data.code}: ${data.msg}`);
        }

        return data;
    }
    catch(err) {
        throw err;
    }
  };

export {fetchApodData, API_URL};
