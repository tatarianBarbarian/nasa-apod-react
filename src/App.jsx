import './App.css'
import { Button, Drawer, Spin } from 'antd';
import DatePicker from './DatePicker';
import 'antd/es/style/default.css';
import 'antd/es/button/style/index.css';
import 'antd/es/drawer/style/index.css';
import 'antd/es/spin/style/index.css';
import { Viewer } from './Viewer';

import { useQuery } from 'react-query';
import { fetchApodData } from './api';
import {useDateReducer} from './useDateReducer';
import { useState } from 'react';
import { formatDate } from './date';

const validateDate = (date) => {
  const minDate = new Date("1995-06-20");
  const today = new Date();
  
  return !(minDate <= date && date <= today);
}

function App() {
  const {
    date, 
    incrementDate, 
    decrementDate, 
    resetDate, 
    setDate, 
    setRandomDate
  } = useDateReducer();
  const formattedDate = formatDate(date);
  const { isLoading, error, data = {} } = useQuery(['apod', formattedDate], fetchApodData);
  const [isDescriptionVisible, setDescriptionVisibile] = useState(false);
  const [isAboutInfoVisible, setAboutInfoVisible] = useState(false);
  const {explanation, title, copyright} = data;
  
  return (
    <div className="App">
      <h1 style={{color: 'white', textAlign: 'center'}}>NASA's Astronomy Picture of the day</h1>
      <div className="controls">
        <Button data-testid="prevBtn" disabled={isLoading} onClick={decrementDate}>Prev</Button>
        <Button data-testid="nextBtn" disabled={isLoading} onClick={incrementDate}>Next</Button>
        <Button data-testid="randomBtn" disabled={isLoading} onClick={setRandomDate}>Random</Button>
        <Button data-testid="aboutBtn" disabled={isLoading} onClick={() => setDescriptionVisibile(true)}>About this pic</Button>
        <Button data-testid="todayBtn" disabled={isLoading} onClick={resetDate}>Today</Button>
        <DatePicker
          data-testid="datepicker" 
          disabled={isLoading}
          disabledDate={validateDate}
          defaultValue={date}
          value={date}
          onChange={setDate}
        />
        <Button onClick={() => setAboutInfoVisible(true)}>‚ùî</Button>
      </div>
      <div style={{textAlign: 'center'}}>
        {isLoading ? <Spin size='large' data-testid="spinner" /> : <Viewer error={error} data={data}/>}
      </div>
      <Drawer
        title="About app"
        placement={'bottom'}
        closable={true}
        onClose={() => setAboutInfoVisible(false)}
        visible={isAboutInfoVisible}
        key={'about app'}
      >
        <p>Based on <a href='https://api.nasa.gov/'>NASA API</a></p>
        <p>Made by <a href='https://github.com/tatarianBarbarian'>Felix Khafizov</a></p>
        <p><a href='https://github.com/tatarianBarbarian/nasa-apod-react'>Source code</a></p>
      </Drawer>
      <Drawer
          title={title}
          placement={'bottom'}
          closable={true}
          onClose={() => setDescriptionVisibile(false)}
          visible={isDescriptionVisible}
          key={'about pic'}
        >
          <div className="description">
            {explanation && <p>{explanation}</p>}
            {copyright && <p>Copyright: {copyright}</p>}
          </div>
      </Drawer>
    </div>
  )
}

export default App
