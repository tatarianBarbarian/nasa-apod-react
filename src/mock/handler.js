import { rest } from "msw";
import { formatDate, getPrevDate } from "../date";
import { API_URL } from "../api";
import { subDays } from "date-fns";

const todayPic = {
    "date": "2022-05-08",
    "explanation": "Today's pic",
    "hdurl": "/src/mock/img/1_hd.jpg",
    "media_type": "image",
    "service_version": "v1",
    "title": "Spiral Galaxy NGC 1512: The Inner Rings",
    "url": "/src/mock/img/1.jpg"
};

const video = {
    "date":"2022-05-09",
    "explanation":"What's that passing in front of the Sun? It looks like a moon, but it can't be Earth's Moon, because it isn't round. It's the Martian moon Phobos.  The featured video was taken from the surface of Mars a month ago by the Perseverance rover. Phobos, at 11.5 kilometers across, is 150 times smaller than Luna (our moon) in diameter, but also 50 times closer to its parent planet.  In fact, Phobos is so close to Mars that it is expected to break up and crash into Mars within the next 50 million years.  In the near term, the low orbit of Phobos results in more rapid solar eclipses than seen from Earth. The featured video is shown in real time -- the transit really took about 40 seconds,as shown. The videographer -- the robotic rover Perseverance (Percy) -- continues to explore Jezero Crater on Mars, searching not only for clues to the watery history of the now dry world, but evidence of ancient microbial life.   New Social Mirror: APOD now available on mastodon",
    "media_type":"video",
    "service_version":"v1",
    "title":"A Martian Eclipse: Phobos Crosses the Sun",
    "url":"https://www.youtube.com/embed/aKK7vS2CHC8?rel=0"
}

const fallbackPic = {
    "copyright": "Carlos Taylor",
    "date": "2022-05-06",
    "explanation": "This cosmic skyscape features glowing gas and dark dust clouds along side the young stars of NGC 3572. A beautiful emission nebula and star cluster it sails far southern skies within the nautical constellation Carina. Stars from NGC 3572 are toward top center in the telescopic frame that would measure about 100 light-years across at the cluster's estimated distance of 9,000 light-years. The visible interstellar gas and dust is part of the star cluster's natal molecular cloud. Dense streamers of material within the nebula, eroded by stellar winds and radiation, clearly trail away from the energetic young stars. They are likely sites of ongoing star formation with shapes reminiscent of the Tadpoles of IC 410 better known to northern skygazers.  In the coming tens to hundreds of millions of years, gas and stars in the cluster will be dispersed though, by gravitational tides and by violent supernova explosions that end the short lives of the massive cluster stars.",
    "hdurl": "/src/mock/img/3_hd.jpg",
    "media_type": "image",
    "service_version": "v1",
    "title": "NGC 3572 and the Southern Tadpoles",
    "url": "/src/mock/img/3.jpg"
}

const other = {
    "copyright": "Jean-Charles Cuillandre",
    "date": "2009-04-13",
    "explanation": "Just fix your camera to a tripod and you too can make an image of graceful trails traced by the stars as planet Earth rotates on its axis.  Making a time lapse video like that shown above may require more effort, though. Made on 2006 October 13 from Mauna Kea, Hawaii, USA, this video nicely captured what you might see of the night sky if you could keep your brain from changing your perceived visual image about every hundredth of a second. Starting from a dark sky and point-like stars, the video demonstrates how stars appear to move over the night as the world turns. Near the center of the developing bull's-eye pattern is Polaris, the North Star.  Visible in the left foreground is the 3.7-meter aperture Canada-France-Hawaii Telescope (CFHT). The red glow that illuminated the CFHT dome near the beginning of the film was created by a car leaving the volcanic summit. The Moon rose about half way through the video and created a white glow that gradually illuminated most of the CFHT dome. The above remarkable time-lapse video was constructed from about 1,000 consecutive frames taken with a digital camera over nearly nine hours.   digg_url = 'http://apod.nasa.gov/apod/ap090413.html'; digg_skin = 'compact';",
    "media_type": "other",
    "service_version": "v1",
    "title": "Star Trails over the Canada-France-Hawaii Telescope"
}

const isTests = import.meta.env.TEST;

export const handler = rest.get(API_URL, (req, res, ctx) => {
    const reqDate = req.url.searchParams.get('date');
    const todayRaw = isTests ? new Date('2022-05-13') : new Date();
    const today = formatDate(todayRaw);

    if (reqDate === today) {
        return res(
            ctx.json(todayPic)
        )
    }

    if (reqDate === formatDate(subDays(todayRaw, 1))) {
        return res(
            ctx.json(video)
        )
    }

    if (reqDate === formatDate(subDays(todayRaw, 2))) {
        return res(
            ctx.json(other)
        )
    }

    return res(ctx.json(fallbackPic))
})
