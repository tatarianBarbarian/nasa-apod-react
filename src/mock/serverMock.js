import { rest, setupWorker } from "msw";
import { formatDate, getPrevDate } from "../date";
import { API_URL } from "../api";

export const handler = rest.get(API_URL, (req, res, ctx) => {
    const reqDate = req.url.searchParams.get('date');
    const today = formatDate(new Date());

    if (reqDate === today) {
        return res(
            ctx.json({
                "date": "2022-05-08",
                "explanation": "Today's pic",
                "hdurl": "/src/mock/img/1_hd.jpg",
                "media_type": "image",
                "service_version": "v1",
                "title": "Spiral Galaxy NGC 1512: The Inner Rings",
                "url": "/src/mock/img/1.jpg"
            })
        )
    }

    if (reqDate === formatDate(getPrevDate(new Date()))) {
        return res(
            ctx.json({
                "copyright": "Tara Mostofi",
                "date": "2022-05-07",
                "explanation": "On certain dates in February, an elusive firefall can be spotted at sunset in Yosemite National Park, when water flows, the weather cooperates and the direction to the setting Sun is just right. Often photographed from vantage points below, at the right moment the park's seasonal Horsetail Fall is isolated in the shadows of the steep walls of El Capitan. Then, still illuminated with rays of reddened sunlight the waterfall briefly takes on a dramatic, fiery appearance. But a Horsetail firefall can be photographed by moonlight too. Even more elusive by moonlight, the firefall effect can also be seen when a bright Moon sets at the right direction along the western horizon. And skies were clear enough for this well-planned imaging of an ephemeral Horsetail firefall, lit by a bright gibbous Moon setting in the early morning hours of April 15.",
                "hdurl": "/src/mock/img/2_hd.jpg",
                "media_type": "image",
                "service_version": "v1",
                "title": "Firefall by Moonlight",
                "url": "/src/mock/img/2.jpg"
            })
        )
    }

    return res(ctx.json({
        "copyright": "Carlos Taylor",
        "date": "2022-05-06",
        "explanation": "This cosmic skyscape features glowing gas and dark dust clouds along side the young stars of NGC 3572. A beautiful emission nebula and star cluster it sails far southern skies within the nautical constellation Carina. Stars from NGC 3572 are toward top center in the telescopic frame that would measure about 100 light-years across at the cluster's estimated distance of 9,000 light-years. The visible interstellar gas and dust is part of the star cluster's natal molecular cloud. Dense streamers of material within the nebula, eroded by stellar winds and radiation, clearly trail away from the energetic young stars. They are likely sites of ongoing star formation with shapes reminiscent of the Tadpoles of IC 410 better known to northern skygazers.  In the coming tens to hundreds of millions of years, gas and stars in the cluster will be dispersed though, by gravitational tides and by violent supernova explosions that end the short lives of the massive cluster stars.",
        "hdurl": "/src/mock/img/3_hd.jpg",
        "media_type": "image",
        "service_version": "v1",
        "title": "NGC 3572 and the Southern Tadpoles",
        "url": "/src/mock/img/3.jpg"
    }))
})

export const worker = setupWorker(handler);
