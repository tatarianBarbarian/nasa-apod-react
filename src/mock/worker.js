import { setupWorker } from "msw";
import { handler } from "./serverMock";

export const worker = setupWorker(handler);
