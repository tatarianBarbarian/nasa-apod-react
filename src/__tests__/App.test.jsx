import { describe } from 'vitest';
import App from '../App';
import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import { QueryClient, QueryClientProvider, setLogger } from 'react-query';
import { setupServer } from 'msw/node';
import { handler } from '../mock/handler';
import { rest } from 'msw';
import userEvent from '@testing-library/user-event'
import { API_URL } from '../api';

const server = setupServer(handler);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

setLogger({
    error: () => {},
    warn: console.warn,
    log: console.log,
})

const Wrapped = () => {
    const queryClient = new QueryClient({
        defaultOptions: {
            queries: {
                retry: false,
            },
        },
    });
    return (
        <QueryClientProvider client={queryClient}>
            <App />
        </QueryClientProvider>
    )
};

describe('App', () => {
    it('should render spinner while fetching data', async () => {
        render(<Wrapped />);

        const spinner = screen.getByTestId('spinner')
        expect(spinner).toBeInTheDocument();
        expect(spinner).toBeVisible();
    });

    it('should disable controls while fetching data', async () => {
        render(<Wrapped />);

        expect(screen.getByTestId('prevBtn')).toBeDisabled();
        expect(screen.getByTestId('nextBtn')).toBeDisabled();
        expect(screen.getByTestId('randomBtn')).toBeDisabled();
        expect(screen.getByTestId('aboutBtn')).toBeDisabled();
        expect(screen.getByTestId('todayBtn')).toBeDisabled();
        expect(screen.getByTestId('datepicker')).toBeDisabled();

        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'));

        expect(screen.getByTestId('prevBtn')).not.toBeDisabled();
        expect(screen.getByTestId('nextBtn')).not.toBeDisabled();
        expect(screen.getByTestId('randomBtn')).not.toBeDisabled();
        expect(screen.getByTestId('aboutBtn')).not.toBeDisabled();
        expect(screen.getByTestId('todayBtn')).not.toBeDisabled();
        expect(screen.getByTestId('datepicker')).not.toBeDisabled();
    });

    it('should display pic info when asked', async () => {
        render(<Wrapped />)

        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'));
        await userEvent.click(screen.getByTestId('aboutBtn'))

        expect(screen.getByText(/today's pic/i)).toBeInTheDocument()
        expect(screen.getByText(/today's pic/i)).toBeVisible()
    })

    it('should display error messages from api', async () => {
        const errorMessage = 'Error happened!';

        server.use(rest.get(API_URL, async (req, res, ctx) => {
            return res(ctx.json({msg: errorMessage, code: 400}))
        }))

        render(<Wrapped />)

        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'))

        const message = screen.getByText(new RegExp(errorMessage, 'i'))

        expect(message).toBeInTheDocument()
        expect(message).toBeVisible()
        expect(message.textContent).toMatchInlineSnapshot('"Error 400: Error happened!"')
    })

    it('should load previous day picture when clicked on prev button', async () => {
        const {container} = render(<Wrapped />)

        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'))
        await userEvent.click(screen.getByTestId('prevBtn'))
        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'))

        expect(container).toMatchSnapshot()
    })

    it.todo('displays disabled previous day button if date is today')

    it('should load next day picture when clicked on next button', async () => {
        const {container} = render(<Wrapped />)

        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'))
        await userEvent.click(screen.getByTestId('prevBtn'))
        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'))
        await userEvent.click(screen.getByTestId('nextBtn'))

        expect(container).toMatchSnapshot()
    })

    it.todo('displays disabled next day button if date is today')

    it('[flaky]should display today photo if today button clicked', async () => {
        const {container} = render(<Wrapped />)

        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'))
        await userEvent.click(screen.getByTestId('prevBtn'))
        await waitForElementToBeRemoved(() => screen.getByTestId('spinner'))
        await userEvent.click(screen.getByTestId('todayBtn'))

        expect(container).toMatchSnapshot()
    })

    it('should display application info if about button clicked', async () => {
        render(<Wrapped />)
        
        await userEvent.click(screen.getByTestId('appInfoBtn'))
        const info = screen.getByTestId('appInfo')
        
        expect(info).toMatchSnapshot()
    })
})
