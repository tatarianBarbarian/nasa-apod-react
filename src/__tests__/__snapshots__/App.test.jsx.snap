// Vitest Snapshot v1

exports[`App > [flaky]should display today photo if today button clicked 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      style="color: white; text-align: center;"
    >
      NASA's Astronomy Picture of the day
    </h1>
    <div
      class="controls"
    >
      <button
        ant-click-animating-without-extra-node="true"
        class="ant-btn ant-btn-default"
        data-testid="prevBtn"
        type="button"
      >
        <span>
          Prev
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="nextBtn"
        type="button"
      >
        <span>
          Next
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="randomBtn"
        type="button"
      >
        <span>
          Random
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="aboutBtn"
        type="button"
      >
        <span>
          About this pic
        </span>
      </button>
      <button
        ant-click-animating-without-extra-node="false"
        class="ant-btn ant-btn-default"
        data-testid="todayBtn"
        type="button"
      >
        <span>
          Today
        </span>
      </button>
      <div
        class="ant-picker"
      >
        <div
          class="ant-picker-input"
        >
          <input
            autocomplete="off"
            data-testid="datepicker"
            placeholder="Select date"
            readonly=""
            size="12"
            title="2022-05-13"
            value="2022-05-13"
          />
          <span
            class="ant-picker-suffix"
          >
            <span
              aria-label="calendar"
              class="anticon anticon-calendar"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
          </span>
          <span
            class="ant-picker-clear"
            role="button"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default"
        data-testid="appInfoBtn"
        type="button"
      >
        <span>
          ❔
        </span>
      </button>
    </div>
    <div
      style="text-align: center;"
    >
      <div
        class="ant-image"
      >
        <img
          class="ant-image-img"
          src="/src/mock/img/1.jpg"
          style="max-height: 80vh; object-fit: contain;"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App > should display application info if about button clicked 1`] = `
<div
  data-testid="appInfo"
>
  <p>
    Based on 
    <a
      href="https://api.nasa.gov/"
    >
      NASA API
    </a>
  </p>
  <p>
    Made by 
    <a
      href="https://github.com/tatarianBarbarian"
    >
      Felix Khafizov
    </a>
  </p>
  <p>
    <a
      href="https://github.com/tatarianBarbarian/nasa-apod-react"
    >
      Source code
    </a>
  </p>
</div>
`;

exports[`App > should load next day picture when clicked on next button 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      style="color: white; text-align: center;"
    >
      NASA's Astronomy Picture of the day
    </h1>
    <div
      class="controls"
    >
      <button
        ant-click-animating-without-extra-node="true"
        class="ant-btn ant-btn-default"
        data-testid="prevBtn"
        type="button"
      >
        <span>
          Prev
        </span>
      </button>
      <button
        ant-click-animating-without-extra-node="false"
        class="ant-btn ant-btn-default"
        data-testid="nextBtn"
        type="button"
      >
        <span>
          Next
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="randomBtn"
        type="button"
      >
        <span>
          Random
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="aboutBtn"
        type="button"
      >
        <span>
          About this pic
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="todayBtn"
        type="button"
      >
        <span>
          Today
        </span>
      </button>
      <div
        class="ant-picker"
      >
        <div
          class="ant-picker-input"
        >
          <input
            autocomplete="off"
            data-testid="datepicker"
            placeholder="Select date"
            readonly=""
            size="12"
            title="2022-05-13"
            value="2022-05-13"
          />
          <span
            class="ant-picker-suffix"
          >
            <span
              aria-label="calendar"
              class="anticon anticon-calendar"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
          </span>
          <span
            class="ant-picker-clear"
            role="button"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default"
        data-testid="appInfoBtn"
        type="button"
      >
        <span>
          ❔
        </span>
      </button>
    </div>
    <div
      style="text-align: center;"
    >
      <div
        class="ant-image"
      >
        <img
          class="ant-image-img"
          src="/src/mock/img/1.jpg"
          style="max-height: 80vh; object-fit: contain;"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App > should load previous day picture when clicked on prev button 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      style="color: white; text-align: center;"
    >
      NASA's Astronomy Picture of the day
    </h1>
    <div
      class="controls"
    >
      <button
        ant-click-animating-without-extra-node="true"
        class="ant-btn ant-btn-default"
        data-testid="prevBtn"
        type="button"
      >
        <span>
          Prev
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="nextBtn"
        type="button"
      >
        <span>
          Next
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="randomBtn"
        type="button"
      >
        <span>
          Random
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="aboutBtn"
        type="button"
      >
        <span>
          About this pic
        </span>
      </button>
      <button
        class="ant-btn ant-btn-default"
        data-testid="todayBtn"
        type="button"
      >
        <span>
          Today
        </span>
      </button>
      <div
        class="ant-picker"
      >
        <div
          class="ant-picker-input"
        >
          <input
            autocomplete="off"
            data-testid="datepicker"
            placeholder="Select date"
            readonly=""
            size="12"
            title="2022-05-12"
            value="2022-05-12"
          />
          <span
            class="ant-picker-suffix"
          >
            <span
              aria-label="calendar"
              class="anticon anticon-calendar"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
          </span>
          <span
            class="ant-picker-clear"
            role="button"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default"
        data-testid="appInfoBtn"
        type="button"
      >
        <span>
          ❔
        </span>
      </button>
    </div>
    <div
      style="text-align: center;"
    >
      <div
        class="ant-image"
      >
        <img
          class="ant-image-img"
          src="/src/mock/img/2.jpg"
          style="max-height: 80vh; object-fit: contain;"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
